src/Orleans/Serialization/SerializationManager.cs

```diff
@@ -2223,6 +2229,8 @@ private static void RegisterSerializationProviders(List<TypeInfo> providerTypes)
                 return;
             }
 
+            externalSerializers.Clear();
+            typeToExternalSerializerDictionary.Clear();
```

Comment:
@jdom This was intentional. If someone explicitly calls RegisterSerializationProviders, presumably they would want to register those specific providers in the order in the list (which would cause the mapping of types to serializers to be invalidated and the list of serializers to be invalid) . Also, since they are called by the initialization routines, then the state should be initialized.